### ISSUE TITLE
Unable to open a new modal in closeAllModals callback unless you use a timeout hack

### ISSUE THREAD

**Issue by rossmartin** [colorfy-software/react-native-modalfy#147]
Hi, thank you for your work on this great project!

With version 3.5.3 we are unable to open a modal from within the callback for `closeAllModals` unless you use a timeout hack.

Here is a snippet that illustrates the problem:

```tsx
closeAllModals(() => {
  // openModal does not work here in the callback
  openModal(name, options)
  // It does work if you wrap openModal in a micro timeout.
  // There seems to be a race condition happening somewhere.
})
```

I think this affects the callback for `closeModal` in the same way also. 

I forked the repo here and added this snippet to the example app to illustrate the issue - https://github.com/rossmartin/react-native-modalfy/commit/a7a2abef15e498ad92795c9e3d7f04cc8874ef63

You should be able to reproduce the issue by the snippet or the forked example app provided.

**Comment by CharlesMangwa** [comment]
hey @rossmartin! thanks for opening this issue. i'll try to have a look at it and see what could be causing this: thanks for providing a repro 👍

**Comment by bryce-mcmath** [comment]
+1

**Comment by CharlesMangwa** [comment]
hey @rossmartin @bryce-mcmath! i just released [`v3.5.4`](https://github.com/colorfy-software/react-native-modalfy/releases/tag/v3.5.4) with a patch for this issue. i couldn't find the root cause but at least you should not have to carry the burden of the timeout hack anymore. let me know if [`v3.5.4`](https://github.com/colorfy-software/react-native-modalfy/releases/tag/v3.5.4) works as expected for you!
